### DISCUSSION TITLE
Sujet / Thema 16 –– Service-Verzeichnis / Catalogue des services

### DISCUSSION THREAD

**Discussion by davidoesch** [GeoUnconference/discussions#38] [Ideen / Idées]
Können wir alle Geodienste der Schweiz (WMTS und andere) in einer grossen Sammlung zusammentragen? (und allenfalls beschreiben wie z.B. [hier](https://nils.ratnaweera.net/posts/2021-12-01-opendata-per-canton/))? 
zb Scraper wms/wmts  Dienste analog wie für COVID https://github.com/openZH/covid_19 umsetzen , eine " Liste" erstellen, die dann zum Beispiel für ein Layertree in einem Viewer  wie in https://github.com/geoadmin/mf-geoadmin3/issues/4668  genutzte werden kann

**Comment by stumpigit** [comment]
Finde ich eine gute Idee, dass es per Scraper möglichst automatisch eine Liste mit Layertree erstellt. Ich würde es auch noch bevorzugen, wenn man dann weitere Kommentare zum Layer und allenfalls ein Rating machen kann. Manchmal ist mir bei gewissen Layern nicht ganz klar, wie die Datengüte ist. 

Zusätzlich stelle ich mir ein "Tag"-System vor, das man zu den Layern vergeben kann. So könnte man vielleicht auch schnell gleiche Tags Zusammenstellen. Dann sieht man schnell, welcher Kanton ähnliche Daten anbietet, was der Bund hat, wo die Unterschiede sind. Ich hoffe, ihr versteht, was ich meine, ansonsten dann am Donnerstag mehr Details... Liebe Grüsse, Christoph

**Reply by davidoesch** [reply]
Tags basierend auf  Metadaten? : zb für Datenherr ( kantonal, privat, bund), GeoIV Nummer, etc?
Rating: Wir hatten mal mit [@andreasamlser ] ein rating pro Datensatz  in opendata.swiss diskutiert ( discourse? )

Die Frage für mich ist:
- nur SCRAPING (dann kommen nur infos rein die wir aus den getCapabilities etc entnehmen können)
- manuelles erstellen der Liste
- SCRAPING und manuelles editieren

**Reply by stumpigit** [reply]
Ich würde scraping mit zusätzlich manuellen Feldern präferieren. Damit könnten die Datenanbieter die Metadaten anpassen, das Scraping übernimmt diese via getCapabilities. In den manuellen Feldern bleiben aber die zusätzlichen Angaben bestehen.

Bezüglich Tags stelle ich mir einen dynamischen Tag-Katalog vor, der mit neuen Tags erweitert werden kann. In den (WMS/WMTS) Metadaten kann direkt auf Tags mit Hashtag verwiesen werden, so dass der Datensatz richtig katalogisiert wird. In einem manuellen Feld können die Tags ausgewählt und dem Datensatz hinzugewiesen werden.

Also Prinzip: Möglichst viele Infos aus GetCapabilities nehmen, aber auch Möglichkeit weitere (v.a. Katalogs- und suchrelevante) Daten abzufüllen.
Ziel: Der Benutzer findet schnell unterschiedliche Datensätze mit ähnlichem Inhalt (im Stile von "Andere Kunden kauften auch...") Unterschiedlich im Sinne von unterschiedlicher geograf. Bezug, Skalierung, Datengüte, ...

**Reply by stumpigit** [reply]
Habe mal drei Wireframes gebastelt, an welchen wir vielleicht morgen einige Ideen besprechen können:

- Ideen für Startseite
- Katalog mit Auflistung der Layers nach Tags / Definierte Hierarchie
- Informationen zum Layer mit Möglichkeit der Community-basierten Erweiterung

![Startseite](https://user-images.githubusercontent.com/9444029/195344624-be1605ea-ea1b-4d42-ae10-d4d76c64d561.png)

![Katalog](https://user-images.githubusercontent.com/9444029/195344616-92404f52-9400-4089-b932-54fe98ec6c19.png)

![Layerinfo](https://user-images.githubusercontent.com/9444029/195344622-849ba5af-159c-439f-9b25-5c1b1f257ef8.png)

**Reply by davidoesch** [reply]
Nice wireframe 
Ich denke wichtig ist vor allem dass wir zuerst die Informationen aggregieren und harmonisieren - und bereitstellen. Wenn wir eine einfach lesbare interpretierbare und vor allem auch automatisch nachgeführte Grundlage haben : dann lassen sich verschiedene Sichten darauf aufbauen.

Es gibt heute schon Verzeichnisse wie Opendata.swiss geocat oder entsprechende plug-ins wie zb in qgis das Swiss-knife.

**Comment by hinjud** [comment]
- Diensttyp: Die Beschränkung auf Darstellungsdienste (WMS/WMTS) empfinde ich als sinnvoll. Downloaddienste (inkl. API-Schnittstellen) könnten zu einem späteren Zeitpunkt miteinbezogen werden.
- Ansatz: 
  - Die Erstellung via WebScraping finde ich eine gute Idee. Im ersten Anlauf würde ich der Einfachheit halber eine Beschränkung auf die Capabilities-Infos vornehmen. 
  - Scraping mit manuellem Editieren: Sobald manuell Inhalte abgefüllt werden müssen, sehe ich einige Hürden bezüglich Konsistenz, Vollständigkeit, Motivation der Bereitsteller etc. Hier müssten entsprechend Vorkehrungen getroffen werden. 
  - Tags basierend auf Metadaten: Wird erwartet, dass diese Metadaten den Capabilities entnommen werden können, gibt es ebenfalls Stolpersteine wie beispielweise die Standardisierung der Inhalte gemäss OGC (gerade bei Bereitstellern, bei denen die Capabilities automatisiert aus bestehenden Metadaten zusammengestellt werden) . Eine Erweiterung der Capabilitiesinhalte z.B. durch Vorgaben via eCH0056, ist zwar denkbar, jedoch vermutlich von wenig Erfolg gekrönt. Gerade die SOLL-Anforderungen werden schon jetzt nur beschränkt umgesetzt. 
  - Listing/Layertree: Bei einem grossen Umfang (und wir wollen ja möglichst viele Einträge :) ) wirkt dies schnell unübersichtlich (Kundenrückmeldungen). Evtl. könnte man dynamische Listings erstellen basierend auf Keywords/Tags o.ä.
- Hosting und Unterhalt: Die Funktion sollte in eine bestehende, bekannte Plattform wie beispielsweise geocat.ch integriert werden. Es gibt schon zu viele lose Plattformen. Zudem muss das Angebot gepflegt und die Applikation etc. gewartet werden. Wieso also nicht etwas bereits existierendes ausbauen und bestehende Ressourcen nutzen.
- Zielgruppe: Das Resultat sollte ein niederschwelliges Angebot sein. Ich würde es als eine eine verpasste Chance sehen, wenn das Endprodukt "nur" technik- bzw. programmiererprobte Personen etc. ansprechen würde.

**Reply by davidoesch** [reply]
Gute Punkte Judith 
- keep it simple es muss niederschwellig sein , das niederschwelligste ist imho eine Tabelle ,csv , excel das gefiltert werden kann
- scraping : ich habe mal einen wms scraper zusammengewurstelt https://github.com/davidoesch/wmschecker war relativ einfach

D

**Comment by hinjud** [comment]
*Problemstellung*
- es gibt zwar viele Daten (räumlich, nicht räumlich), jedoch 
   - sind diese auf verschiedene Plattformen verteilen 
   - z.T. fehlende oder mangelnde Metadaten bzw. Dokumentation
- unterschieldiche Nutzergruppen:
  -->  Aktuelle Plattformen tragen diesem Umstand zu wenig Rechnung

*Ziel*
- Möglichkeit für eine nutzergruppenorientierte Datensuche, wobei Daten sämlicher potentieller Bereitsteller berücksichtigt werden --> partizipativ, bottom-up

**Comment by hinjud** [comment]
Zur Realisierung der Vision ist die Klärung folgender Punkte essentiell:
1. Bereitstellung
   - Inhaltsspektrum: Daten, Services (inkl. funktionaler Dienste), APIs
   - Metadaten: 
     - Hol-/Bringschuld --> Scraping, Scraping + manuelle Erweiterung
     - Forderung: Einheitlicher Layertree basierend auf einheitlichen Tags/Keywords (schweizweit) 
   - QS für manuelle Erweiterungen (restliche Daten unterliegen bereits einem Kontrollmechanismus) --> community-driven
   - Möglichkeit der Öffnung gegenüber Datenprovidern aus Privatwirtschaft, Forschung etc.
2. Suche/Resultat:
   - Eingabe von Abfragekriterien inkl. Filterung (inhaltlich/thematisch, zeitlich, räumlich, Nutzungsvoraussetzungen, Qualitätsmerkmale)
   - Nutzung der einheitlichen Tags/Keywords
3. Nutzerprofile:
    - Bulk-User --> Nutzung/Bezug von Rohdaten im grossen Stil
    - Entwickler --> Nutzung von Daten via Services
    - Gelegenheitsnutzer --> Nutzung von Daten via Applikation
5. Technik: aus unserer Sicht vermutlich lösbar, wobei ein etappiertes Vorgehen sinnvoll wäre
6. Medium/Plattform: 
   - Integration in eine bestehende Lösung --> Nutzung bereits bestehender Ressourcen, Sicherung des Fortbestandes und der kontinuierlichen Pflege
   - Trägerschaft

![image](https://user-images.githubusercontent.com/4510815/196648625-0c614ed6-0d6c-4100-adea-6cf8202980a9.png)

**Comment by hinjud** [comment]
*Herausforderungen*
- Akzeptanz über alle Nutzergruppen --> 
- Qualität der Daten vs. Bereitstellungsgeschwindigkeit
- ...

**Comment by rastrau** [comment]
fyi: The group that has compiled the afore-listed results comprised @hinjud, @stumpigit, Martin Probst and Peter Robineau.

**Comment by davidoesch** [comment]
Dear group : a  POC https://github.com/davidoesch/geoservice_harvester_poc of a "Service Harvester" has been established and we will bring it an operational level together with an FHNW project ....

**Reply by hinjud** [reply]
That's great! Thank you very much for your efforts!

**Reply by stumpigit** [reply]
Very cool and useful stuff already, thank you very much! What are the further plans? Maybe I can support something somewhere?

**Reply by davidoesch** [reply]
Plans are in issues documented.
FHNW will receive a grant to work on them.

I'm try to fork this project to the https://github.com/geoadmin domain.

And then it is to us , the community, to add the services

**Comment by davidoesch** [comment]
Dear all 
short update on https://github.com/davidoesch/geoservice_harvester_poc 

Done
- [x] 12 Kantons are now daily harvested https://github.com/davidoesch/geoservice_harvester_poc#current-status-provider--services
- [x] simple search and visualiser unter  https://davidoesch.github.io/geoservice_harvester_poc/ 
- [x] CSV [All SERVICES attributes](https://github.com/davidoesch/geoservice_harvester_poc/blob/main/data/geoservices_CH.csv)
- [x] CSV [All DATA attributes](https://github.com/davidoesch/geoservice_harvester_poc/blob/main/data/geodata_CH.csv)

to Do
- [x] Do the other https://github.com/davidoesch/geoservice_harvester_poc#current-status-provider--services any hellp PR Welcome
- [x] contact AI AR geoinfo.ch reagarding  OGC Compliance issue

**Comment by davidoesch** [comment]
Dear all

All cantons are now integrated:
https://github.com/davidoesch/geoservice_harvester_poc

And I wrote a short blog post about it
 https://www.linkedin.com/pulse/geoharvesting-unexploiting-seo-treasures-geoservices-david-oesch

**Comment by davidoesch** [comment]
Dear All
Wir haben nun den POC mit der FHNW begonnen: Projektdoku und Fortschritt unter https://github.com/FHNW-IVGI/Geoharvester dokumentiert...

**Comment by davidoesch** [comment]
you can now search 21'000 datasets from swiss gov and drag and drop the desired Layer /Dataset as WMS WMTS WFS directly to your QGIS or ArcGISPro Desktop Client, feedback welcome : test it: https://davidoesch.github.io/geoservice_harvester_poc/
